src/cmd/troubleshoot/image_pullable.go

```diff
@@ -90,67 +104,111 @@ func checkActiveGateImagePullable(troubleshootCtx *troubleshootContext) error {
 	if err != nil {
 		return err
 	}
-	logInfof("using '%s' on '%s' with version '%s' as %s image", componentImage, componentRegistry, componentVersion, componentName)
+	logInfof("using '%s' on '%s' with version '%s' as %s image", componentImageInfo.image, componentImageInfo.registry, componentImageInfo.version, componentName)
 
 	imageWorks := false
 
 	// parse docker config
 	var result Auths
-	json.Unmarshal([]byte(pullSecret), &result)
+	if err := json.Unmarshal([]byte(pullSecret), &result); err != nil {
+		logErrorf("invalid pull secret, could not unmarshal to JSON: %v", err)
+		return nil
+	}
 
 	for registry, endpoint := range result.Auths {
 		logInfof("checking images for registry '%s'", registry)
 
 		apiToken := base64.StdEncoding.EncodeToString([]byte(endpoint.Username + ":" + endpoint.Password))
 
-		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/", "Basic", apiToken); err != nil {
-			logErrorf("registry '%s' unreachable", registry)
+		if err := registryAvailable(httpClient, registry+"/v2/", apiToken); err != nil {
+			logErrorf("%v", err)
 			continue
-		} else {
-			if statusCode != 200 {
-				logErrorf("registry '%s' unreachable (%d)", registry, statusCode)
-				continue
-			} else {
-				logInfof("registry '%s' is accessible", registry)
-			}
 		}
 
-		if statusCode, err := connectToDockerRegistry(httpClient, "HEAD", "https://"+registry+"/v2/"+componentImage+"/manifests/"+componentVersion, "Basic", apiToken); err != nil {
-			logErrorf("registry '%s' unreachable", registry)
+		if err := imageAvailable(httpClient, "https://"+registry+"/v2/"+componentImageInfo.image+"/manifests/"+componentImageInfo.version, apiToken); err != nil {
+			logErrorf("cannot pull image '%s' with version '%s' from registry '%s': %v", componentImageInfo.image, componentImageInfo.version, registry, err)
 			continue
 		} else {
-			if statusCode != 200 {
-				logErrorf("image '%s' with version '%s' not found on registry '%s'", componentImage, componentVersion, registry)
-				continue
-			} else {
-				logInfof("image '%s' with version '%s' exists on registry '%s", componentImage, componentVersion, registry)
-			}
+			logInfof("image '%s' with version '%s' exists on registry '%s", componentImageInfo.image, componentImageInfo.version, registry)
 		}
 
 		imageWorks = true
 	}
 
 	if imageWorks {
-		logOkf("%s image '%s' found", componentName, componentRegistry+"/"+componentImage)
+		logOkf("%s image '%s' found", componentName, componentImageInfo.registry+"/"+componentImageInfo.image)
 	} else {
-		return fmt.Errorf("%s image '%s' missing", componentName, componentRegistry+"/"+componentImage)
+		return fmt.Errorf("%s image '%s' missing", componentName, componentImageInfo.registry+"/"+componentImageInfo.image)
+	}
+	return nil
+}
+
+func checkCustomModuleImagePullable(httpClient *http.Client, componentName string, pullSecret string, codeModulesImage string) error {
```

Comment:
there is a docker library we already use in `installer/image` that could be used to do this. Would save us the need to re-implement this functionality if we can use that.

Reply:
will note that down with all the other findings.